### ISSUE TITLE
libvirt-xen failing to update/install without libnbd (Manjaro with pamac)

### ISSUE THREAD

**Issue by peter-kehl** [ahmubashshir/pkgbuilds#10]
Hi Ahmad Hasan,

Thank you for maintaining so many PKG builds.

Summary:
1. 5 years old Manjaro, kept very up to date (almost exclusively with GNOME "Add/Remove Software", not with `pacman`).
3. I did have `libvirt-xen` successfully installed already (with "Add/Remove Software"). But the most recent update failed, saying that it needed `libnbd`.
4. Side note: Before the most recent `libvirt-xen` update, I've uninstalled all Haskell libraries and Haskell itself (using "Add/Remove Software"). Now I am surprised that I didn't get any error (when removing Haskell), because re-installing `libvirt-xen` put Haskell back.

I do know that `AUR` and `archlinux.org` is primarily for Arch, and that Manjaro is somewhat non-standard, and that the main supported method of installing/updating packages is with `pacman`.

However, I don't know how to easily reproduce with pacman (I don't know how to enable/add AUR to pacman's config - but I do have it enabled for "Add/Remove Software"). If you'd like me to troubleshoot, send steps/commands to run, please.

Advance apologies if any of the below logs are irrelevant, or too short (because GitHub was giving me "body is too long (maximum is 65536 characters)"). Again, I'm open to investigating, feel free to send me steps/commands.

![image](https://github.com/ahmubashshir/pkgbuilds/assets/4270240/e8ed4075-12f3-46b0-93b1-03341c5635ce)

```
sudo pacman -S libvirt-xen
error: target not found: libvirt-xen
```

```
sudo pamac install libvirt-xen
Warning: libvirt-xen is only available from AUR
Preparing...
Checking libvirt-xen dependencies...
Checking xen dependencies...
Resolving dependencies...
Checking inter-conflicts...

To install (227):
  ghc-libs
  ...
Apply transaction ? [e/y/N]
...
  LD      arch/x86/built_in.o
  LD      prelink.o
ld    -melf_x86_64 --no-warn-rwx-segments -T arch/x86/xen.lds -N prelink.o --build-id=sha1 \
    ./common/symbols-dummy.o -o ./.xen-syms.0
  HOSTCC  arch/x86/efi/mkreloc
nm -pa --format=sysv ./.xen-syms.0 \
	| ./tools/symbols --all-symbols --sort-by-name --sysv --sort \
	>./.xen-syms.0.S
make -f ./Rules.mk obj=. ./.xen-syms.0.o
  HOSTCC  arch/x86/boot/mkelf32
  CC      .xen-syms.0.o
ld -mi386pep --no-warn-rwx-segments --subsystem=10 --image-base=0xffff82d040000000 --stack=0,0 --heap=0,0 --section-alignment=0x200000 --file-alignment=0x20 --major-image-version=4 --minor-image-version=17 --major-os-version=2 --minor-os-version=0 --major-subsystem-version=2 --minor-subsystem-version=0 --no-insert-timestamp --build-id=sha1 -T arch/x86/efi.lds -N prelink.o  ./common/symbols-dummy.o -b pe-x86-64 arch/x86/efi/buildid.o -o ./.xen.efi.0xffff82d040000000.0 && :
ld    -melf_x86_64 --no-warn-rwx-segments -T arch/x86/xen.lds -N prelink.o --build-id=sha1 \
    ./.xen-syms.0.o -o ./.xen-syms.1
nm -pa --format=sysv ./.xen-syms.1 \
	| ./tools/symbols --all-symbols --sort-by-name --sysv --sort --error-dup \
	>./.xen-syms.1.S
make -f ./Rules.mk obj=. ./.xen-syms.1.o
  CC      .xen-syms.1.o
ld    -melf_x86_64 --no-warn-rwx-segments -T arch/x86/xen.lds -N prelink.o --build-id=sha1 \
    --orphan-handling=warn ./.xen-syms.1.o -o xen-syms
nm -pa --format=sysv ./xen-syms \
	| ./tools/symbols --all-symbols --xensyms --sysv --sort \
	>./xen-syms.map
rm -f ./.xen-syms.[0-9]* ./..xen-syms.[0-9]*
/bin/sh ./tools/check-endbr.sh xen-syms
: ./.xen.efi.0xffff82d040000000.0 >./.xen.efi.0r.S
nm -pa --format=sysv ./.xen.efi.0xffff82d040000000.0 \
	| ./tools/symbols --all-symbols --sort-by-name --sysv --sort >./.xen.efi.0s.S
make -f ./Rules.mk obj=. .xen.efi.0r.o .xen.efi.0s.o
  CC      .xen.efi.0r.o
  CC      .xen.efi.0s.o
ld -mi386pep --no-warn-rwx-segments --subsystem=10 --image-base=0xffff82d040000000 --stack=0,0 --heap=0,0 --section-alignment=0x200000 --file-alignment=0x20 --major-image-version=4 --minor-image-version=17 --major-os-version=2 --minor-os-version=0 --major-subsystem-version=2 --minor-subsystem-version=0 --no-insert-timestamp --build-id=sha1 -T arch/x86/efi.lds -N prelink.o ./.xen.efi.0r.o ./.xen.efi.0s.o -b pe-x86-64 arch/x86/efi/buildid.o -o ./.xen.efi.0xffff82d040000000.1 && :
: ./.xen.efi.0xffff82d040000000.1 >./.xen.efi.1r.S
nm -pa --format=sysv ./.xen.efi.0xffff82d040000000.1 \
	| ./tools/symbols --all-symbols --sort-by-name --sysv --sort >./.xen.efi.1s.S
make -f ./Rules.mk obj=. .xen.efi.1r.o .xen.efi.1s.o
  CC      .xen.efi.1r.o
  CC      .xen.efi.1s.o
ld -mi386pep --no-warn-rwx-segments --subsystem=10 --image-base=0xffff82d040000000 --stack=0,0 --heap=0,0 --section-alignment=0x200000 --file-alignment=0x20 --major-image-version=4 --minor-image-version=17 --major-os-version=2 --minor-os-version=0 --major-subsystem-version=2 --minor-subsystem-version=0 --no-insert-timestamp --build-id=sha1 -T arch/x86/efi.lds -N prelink.o \
      ./.xen.efi.1r.o ./.xen.efi.1s.o --orphan-handling=warn -b pe-x86-64 arch/x86/efi/buildid.o -o xen.efi
nm -pa --format=sysv ./xen.efi \
	| ./tools/symbols --all-symbols --xensyms --sysv --sort >./xen.efi.map
rm -f ./.xen.efi.[0-9]* ./..xen.efi.[0-9]*
/bin/sh ./tools/check-endbr.sh xen.efi
arch/x86/boot/mkelf32 --notes xen-syms ./.xen.elf32 0x200000 \
               `nm xen-syms | sed -ne 's/^\([^ ]*\) . __2M_rwdata_end$/0x\1/p'`
od -t x4 -N 8192 ./.xen.elf32  | grep 1badb002 > /dev/null || \
	{ echo "No Multiboot1 header found" >&2; false; }
od -t x4 -N 32768 ./.xen.elf32 | grep e85250d6 > /dev/null || \
	{ echo "No Multiboot2 header found" >&2; false; }
mv ./.xen.elf32 xen
gzip -n -f -9 < xen > xen.gz.new
mv xen.gz.new xen.gz
[ -d /var/cache/private/pamac/xen/src/xen/dist/install/boot ] || install -d -m0755 -p /var/cache/private/pamac/xen/src/xen/dist/install/boot
install -m0644 -p xen.gz /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.17.3-pre-arch.gz
ln -f -s xen-4.17.3-pre-arch.gz /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.17.gz
ln -f -s xen-4.17.3-pre-arch.gz /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.gz
ln -f -s xen-4.17.3-pre-arch.gz /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen.gz
[ -d "/var/cache/private/pamac/xen/src/xen/dist/install/usr/lib/debug" ] || install -d -m0755 -p /var/cache/private/pamac/xen/src/xen/dist/install/usr/lib/debug
install -m0644 -p xen-syms /var/cache/private/pamac/xen/src/xen/dist/install/usr/lib/debug/xen-syms-4.17.3-pre-arch
install -m0644 -p xen-syms.map /var/cache/private/pamac/xen/src/xen/dist/install/usr/lib/debug/xen-syms-4.17.3-pre-arch.map
install -m0644 -p .config /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.17.3-pre-arch.config
if [ -r xen.efi -a -n '/boot' ]; then \
	[ -d /var/cache/private/pamac/xen/src/xen/dist/install/boot ] || install -d -m0755 -p /var/cache/private/pamac/xen/src/xen/dist/install/boot; \
	install -m0644 -p xen.efi /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.17.3-pre-arch.efi; \
	if [ -e xen.efi.map ]; then \
		install -m0644 -p xen.efi.map /var/cache/private/pamac/xen/src/xen/dist/install/usr/lib/debug/xen-4.17.3-pre-arch.efi.map; \
	fi; \
	ln -sf xen-4.17.3-pre-arch.efi /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.17.efi; \
	ln -sf xen-4.17.3-pre-arch.efi /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen-4.efi; \
	ln -sf xen-4.17.3-pre-arch.efi /var/cache/private/pamac/xen/src/xen/dist/install/boot/xen.efi; \
	if [ -n '/boot' -a -n '' ]; then \
		 \
		install -m0644 -p xen.efi /var/cache/private/pamac/xen/src/xen/dist/install/boot/efi//xen-4.17.3-pre-arch.efi; \
	elif [ "/var/cache/private/pamac/xen/src/xen/dist/install" = "dist/install" ]; then \
		echo 'EFI installation only partially done (EFI_VENDOR not set)' >&2; \
	fi; \
fi
make[1]: Leaving directory '/var/cache/private/pamac/xen/src/xen/xen'
==> ERROR: A failure occurred in build().
    Aborting...
Finished with result: exit-code
Main processes terminated with: code=exited/status=4
Service runtime: 4min 51.263s
CPU time consumed: 7min 40.898s
```

```
Preparing...
Synchronizing package databases...
Refreshing core.db...
Refreshing extra.db...
Refreshing community.db...
Refreshing multilib.db...
Refreshing pritunl.db...
Refreshing core.files...
Refreshing extra.files...
Refreshing community.files...
Refreshing multilib.files...
Refreshing pritunl.files...
Refreshing AUR...
Warning: glide: local (0.13.3-3) is newer than extra (0.6.0-1)
Warning: gnome-wallpapers: local (20201023-1) is newer than extra (20201020-2)
Nothing to do.
Transaction successfully finished.

Preparing...
Cloning libvirt-xen build files...
Generating libvirt-xen information...
Checking libvirt-xen dependencies...
Checking xen dependencies...
Resolving dependencies...
Checking inter-conflicts...
Cloning xen build files...
Generating xen information...

Building xen...
==> Making package: xen 4.17.3pre-1 (Sun 19 Nov 2023 11:23:38 AM)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Cloning xen git repo...
Cloning into bare repository '/var/tmp/pamac-build-pkehl/xen/xen'...
  -> Found efi-xen.cfg
...
==> Verifying source file signatures with gpg...
    libvirt-9.9.0.tar.xz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting libvirt-9.9.0.tar.xz with bsdtar
==> Starting prepare()...
==> Starting build()...
+ exec meson setup --prefix /usr --libexecdir lib --sbindir bin --buildtype plain --auto-features enabled --wrap-mode nodownload -D b_lto=true -D b_pie=true -D python.bytecompile=1 build libvirt-9.9.0 --libexecdir=lib/libvirt -Drunstatedir=/run -Dqemu_user=libvirt-qemu -Dqemu_group=libvirt-qemu -Dnetcf=disabled -Dopenwsman=disabled -Dapparmor=disabled -Dapparmor_profiles=disabled -Dselinux=disabled -Dwireshark_dissector=disabled -Ddriver_bhyve=disabled -Ddriver_hyperv=disabled -Ddriver_libxl=enabled -Ddriver_vz=disabled -Dsanlock=disabled -Dsecdriver_apparmor=disabled -Dsecdriver_selinux=disabled -Dstorage_vstorage=disabled -Ddtrace=disabled -Dnumad=disabled -Dstorage_zfs=enabled -Dstorage_rbd=disabled
The Meson build system
Version: 1.2.3
Source dir: /var/tmp/pamac-build-pkehl/libvirt-xen/src/libvirt-9.9.0
Build dir: /var/tmp/pamac-build-pkehl/libvirt-xen/src/build
Build type: native build
Project name: libvirt
Project version: 9.9.0
C compiler for the host machine: cc (gcc 13.2.1 "cc (GCC) 13.2.1 20230801")
C linker for the host machine: cc ld.bfd 2.41.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Configuring configmake.h using configuration
Checking for size of "ptrdiff_t" : 8 
Checking for size of "size_t" : 8 
Compiler for C supports arguments -fasynchronous-unwind-tables: YES 
Compiler for C supports arguments -fexceptions: YES 
Compiler for C supports arguments -fipa-pure-const: YES 
Compiler for C supports arguments -fno-common: YES 
Compiler for C supports arguments -Wabsolute-value: YES 
Compiler for C supports arguments -Waddress: YES 
Compiler for C supports arguments -Waddress-of-packed-member: YES 
Compiler for C supports arguments -Waggressive-loop-optimizations: YES 
Compiler for C supports arguments -Walloc-size-larger-than=9223372036854775807: YES 
Compiler for C supports arguments -Walloca: YES 
Compiler for C supports arguments -Warray-bounds=2: YES 
Compiler for C supports arguments -Wattribute-alias=2: YES 
Compiler for C supports arguments -Wattribute-warning: YES 
Compiler for C supports arguments -Wattributes: YES 
Compiler for C supports arguments -Wbool-compare: YES 
Compiler for C supports arguments -Wbool-operation: YES 
Compiler for C supports arguments -Wbuiltin-declaration-mismatch: YES 
Compiler for C supports arguments -Wbuiltin-macro-redefined: YES 
Compiler for C supports arguments -Wcannot-profile: YES 
Compiler for C supports arguments -Wcast-align: YES 
Compiler for C supports arguments -Wcast-align=strict: YES 
Compiler for C supports arguments -Wno-cast-function-type: YES 
Compiler for C supports arguments -Wchar-subscripts: YES 
Compiler for C supports arguments -Wclobbered: YES 
Compiler for C supports arguments -Wcomment: YES 
Compiler for C supports arguments -Wcomments: YES 
Compiler for C supports arguments -Wcoverage-mismatch: YES 
Compiler for C supports arguments -Wcpp: YES 
Compiler for C supports arguments -Wdangling-else: YES 
Compiler for C supports arguments -Wdate-time: YES 
Compiler for C supports arguments -Wdeclaration-after-statement: YES 
Compiler for C supports arguments -Wdeprecated-declarations: YES 
Compiler for C supports arguments -Wdesignated-init: YES 
Compiler for C supports arguments -Wdiscarded-array-qualifiers: YES 
Compiler for C supports arguments -Wdiscarded-qualifiers: YES 
Compiler for C supports arguments -Wdiv-by-zero: YES 
Compiler for C supports arguments -Wduplicated-cond: YES 
Compiler for C supports arguments -Wduplicate-decl-specifier: YES 
Compiler for C supports arguments -Wempty-body: YES 
Compiler for C supports arguments -Wendif-labels: YES 
Compiler for C supports arguments -Wexpansion-to-defined: YES 
Compiler for C supports arguments -Wformat-contains-nul: YES 
Compiler for C supports arguments -Wformat-extra-args: YES 
Compiler for C supports arguments -Wno-format-nonliteral: YES 
Compiler for C supports arguments -Wformat-overflow=2: YES 
Compiler for C supports arguments -Wformat-security: YES 
Compiler for C supports arguments -Wno-format-truncation: YES 
Compiler for C supports arguments -Wformat-y2k: YES 
Compiler for C supports arguments -Wformat-zero-length: YES 
Compiler for C supports arguments -Wframe-address: YES 
Compiler for C supports arguments -Wframe-larger-than=2048: YES 
Compiler for C supports arguments -Wfree-nonheap-object: YES 
Compiler for C supports arguments -Whsa: YES 
Compiler for C supports arguments -Wif-not-aligned: YES 
Compiler for C supports arguments -Wignored-attributes: YES 
Compiler for C supports arguments -Wignored-qualifiers: YES 
Compiler for C supports arguments -Wimplicit: YES 
Compiler for C supports arguments -Wimplicit-fallthrough=5: YES 
Compiler for C supports arguments -Wimplicit-function-declaration: YES 
Compiler for C supports arguments -Wimplicit-int: YES 
Compiler for C supports arguments -Wincompatible-pointer-types: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Winline: YES 
Compiler for C supports arguments -Wint-conversion: YES 
Compiler for C supports arguments -Wint-in-bool-context: YES 
Compiler for C supports arguments -Wint-to-pointer-cast: YES 
Compiler for C supports arguments -Winvalid-memory-model: YES 
Compiler for C supports arguments -Winvalid-pch: YES 
Compiler for C supports arguments -Wjump-misses-init: YES 
Compiler for C supports arguments -Wlogical-not-parentheses: YES 
Compiler for C supports arguments -Wlogical-op: YES 
Compiler for C supports arguments -Wmain: YES 
Compiler for C supports arguments -Wmaybe-uninitialized: YES 
Compiler for C supports arguments -Wmemset-elt-size: YES 
Compiler for C supports arguments -Wmemset-transposed-args: YES 
Compiler for C supports arguments -Wmisleading-indentation: YES 
Compiler for C supports arguments -Wmissing-attributes: YES 
Compiler for C supports arguments -Wmissing-braces: YES 
Compiler for C supports arguments -Wmissing-declarations: YES 
Compiler for C supports arguments -Wmissing-field-initializers: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Wmissing-parameter-type: YES 
Compiler for C supports arguments -Wmissing-profile: YES 
Compiler for C supports arguments -Wmissing-prototypes: YES 
Compiler for C supports arguments -Wmultichar: YES 
Compiler for C supports arguments -Wmultistatement-macros: YES 
Compiler for C supports arguments -Wnarrowing: YES 
Compiler for C supports arguments -Wnested-externs: YES 
Compiler for C supports arguments -Wnonnull: YES 
Compiler for C supports arguments -Wnonnull-compare: YES 
Compiler for C supports arguments -Wnormalized=nfc: YES 
Compiler for C supports arguments -Wnull-dereference: YES 
Compiler for C supports arguments -Wodr: YES 
Compiler for C supports arguments -Wold-style-declaration: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Wopenmp-simd: YES 
Compiler for C supports arguments -Woverflow: YES 
Compiler for C supports arguments -Woverride-init: YES 
Compiler for C supports arguments -Wpacked-bitfield-compat: YES 
Compiler for C supports arguments -Wpacked-not-aligned: YES 
Compiler for C supports arguments -Wparentheses: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Compiler for C supports arguments -Wpointer-compare: YES 
Compiler for C supports arguments -Wpointer-sign: YES 
Compiler for C supports arguments -Wpointer-to-int-cast: YES 
Compiler for C supports arguments -Wpragmas: YES 
Compiler for C supports arguments -Wpsabi: YES 
Compiler for C supports arguments -Wrestrict: YES 
Compiler for C supports arguments -Wreturn-local-addr: YES 
Compiler for C supports arguments -Wreturn-type: YES 
Compiler for C supports arguments -Wscalar-storage-order: YES 
Compiler for C supports arguments -Wsequence-point: YES 
Compiler for C supports arguments -Wshadow: YES 
Compiler for C supports arguments -Wshift-count-negative: YES 
Compiler for C supports arguments -Wshift-count-overflow: YES 
Compiler for C supports arguments -Wshift-negative-value: YES 
Compiler for C supports arguments -Wshift-overflow=2: YES 
Compiler for C supports arguments -Wno-sign-compare: YES 
Compiler for C supports arguments -Wsizeof-array-argument: YES 
Compiler for C supports arguments -Wsizeof-pointer-div: YES 
Compiler for C supports arguments -Wsizeof-pointer-memaccess: YES 
Compiler for C supports arguments -Wstrict-aliasing: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wstringop-overflow=2: YES 
Compiler for C supports arguments -Wstringop-truncation: YES 
Compiler for C supports arguments -Wsuggest-attribute=cold: YES 
Compiler for C supports arguments -Wno-suggest-attribute=const: YES 
Compiler for C supports arguments -Wsuggest-attribute=format: YES 
Compiler for C supports arguments -Wsuggest-attribute=noreturn: YES 
Compiler for C supports arguments -Wno-suggest-attribute=pure: YES 
Compiler for C supports arguments -Wsuggest-final-methods: YES 
Compiler for C supports arguments -Wsuggest-final-types: YES 
Compiler for C supports arguments -Wswitch: YES 
Compiler for C supports arguments -Wswitch-bool: YES 
Compiler for C supports arguments -Wswitch-enum: YES 
Compiler for C supports arguments -Wswitch-unreachable: YES 
Compiler for C supports arguments -Wsync-nand: YES 
Compiler for C supports arguments -Wtautological-compare: YES 
Compiler for C supports arguments -Wtrampolines: YES 
Compiler for C supports arguments -Wtrigraphs: YES 
Compiler for C supports arguments -Wtype-limits: YES 
Compiler for C supports arguments -Wno-typedef-redefinition: NO 
Compiler for C supports arguments -Wuninitialized: YES 
Compiler for C supports arguments -Wunknown-pragmas: YES 
Compiler for C supports arguments -Wunused: YES 
Compiler for C supports arguments -Wunused-but-set-parameter: YES 
Compiler for C supports arguments -Wunused-but-set-variable: YES 
Compiler for C supports arguments -Wunused-const-variable=2: YES 
Compiler for C supports arguments -Wunused-function: YES 
Compiler for C supports arguments -Wunused-label: YES 
Compiler for C supports arguments -Wunused-local-typedefs: YES 
Compiler for C supports arguments -Wunused-parameter: YES 
Compiler for C supports arguments -Wunused-result: YES 
Compiler for C supports arguments -Wunused-value: YES 
Compiler for C supports arguments -Wunused-variable: YES 
Compiler for C supports arguments -Wvarargs: YES 
Compiler for C supports arguments -Wvariadic-macros: YES 
Compiler for C supports arguments -Wvector-operation-performance: YES 
Compiler for C supports arguments -Wvla: YES 
Compiler for C supports arguments -Wvolatile-register-var: YES 
Compiler for C supports arguments -Wwrite-strings: YES 
Compiler for C supports arguments -fstack-protector-strong: YES 
First supported argument: -fstack-protector-strong
Checking if "-Wdouble-promotion" compiles: YES 
Compiler for C supports arguments -Wsuggest-attribute=format: YES (cached)
Compiler for C supports arguments -Wframe-larger-than=262144: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -Wl,-z,now: YES 
Compiler for C supports link arguments -Wl,-z,nodelete: YES 
Compiler for C supports link arguments -Wl,-z,defs: YES 
Compiler for C supports link arguments -Wl,--no-copy-dt-needed-entries: YES 
Compiler for C supports link arguments -Wl,-export-dynamic: YES 
First supported link argument: -Wl,-export-dynamic
Checking for function "elf_aux_info" : NO 
Checking for function "explicit_bzero" : YES 
Checking for function "fallocate" : YES 
Checking for function "getauxval" : YES 
Checking for function "getegid" : YES 
Checking for function "geteuid" : YES 
Checking for function "getgid" : YES 
Checking for function "getifaddrs" : YES 
Checking for function "getmntent_r" : YES 
Checking for function "getpwuid_r" : YES 
Checking for function "getrlimit" : YES 
Checking for function "getuid" : YES 
Checking for function "getutxid" : YES 
Checking for function "if_indextoname" : YES 
Checking for function "mmap" : YES 
Checking for function "newlocale" : YES 
Checking for function "pipe2" : YES 
Checking for function "posix_fallocate" : YES 
Checking for function "posix_memalign" : YES 
Checking for function "prlimit" : YES 
Checking for function "sched_setscheduler" : YES 
Checking for function "setgroups" : YES 
Checking for function "setrlimit" : YES 
Checking for function "symlink" : YES 
Checking for function "sysctlbyname" : NO 
Checking for function "__lxstat" : YES 
Checking for function "__lxstat64" : YES 
Checking for function "__xstat" : YES 
Checking for function "__xstat64" : YES 
Checking for function "lstat" : YES 
Checking for function "lstat64" : YES 
Checking for function "stat" : YES 
Checking for function "stat64" : YES 
Header "sys/stat.h" has symbol "__lxstat" : NO 
Header "sys/stat.h" has symbol "__lxstat64" : NO 
Header "sys/stat.h" has symbol "__xstat" : NO 
Header "sys/stat.h" has symbol "__xstat64" : NO 
Header "sys/stat.h" has symbol "lstat" : YES 
Header "sys/stat.h" has symbol "lstat64" : NO 
Header "sys/stat.h" has symbol "stat" : YES 
Header "sys/stat.h" has symbol "stat64" : NO 
Check usable header "asm/hwcap.h" : NO 
Check usable header "ifaddrs.h" : YES 
Check usable header "libtasn1.h" : YES 
Check usable header "linux/kvm.h" : YES 
Check usable header "mntent.h" : YES 
Check usable header "net/ethernet.h" : YES 
Check usable header "net/if.h" : YES 
Check usable header "pty.h" : YES 
Check usable header "pwd.h" : YES 
Check usable header "sched.h" : YES 
Check usable header "sys/auxv.h" : YES 
Check usable header "sys/ioctl.h" : YES 
Check usable header "sys/mount.h" : YES 
Check usable header "sys/syscall.h" : YES 
Check usable header "sys/ucred.h" : NO 
Check usable header "syslog.h" : YES 
Check usable header "util.h" : NO 
Check usable header "xlocale.h" : NO 
Check usable header "linux/sockios.h" : YES 
Header "endian.h" has symbol "htole64" : YES 
Header "unistd.h" has symbol "SEEK_HOLE" : YES 
Header "net/if_dl.h" has symbol "link_addr" : NO 
Header "sched.h" has symbol "cpu_set_t" : YES 
Header "sys/syscall.h" has symbol "SYS_pidfd_open" : YES 
Header "net/if_bridgevar.h" has symbol "BRDGSFD" : NO 
Header "sys/cpuset.h" has symbol "cpuset_getaffinity" : NO 
Header "mach/clock.h" has symbol "clock_serv_t" : NO 
Checking for type "struct ifreq" : YES 
Checking for type "struct sockpeercred" : NO 
Checking whether type "struct ifreq" has member "ifr_newname" : YES 
Checking whether type "struct ifreq" has member "ifr_ifindex" : YES 
Checking whether type "struct ifreq" has member "ifr_index" : NO 
Checking whether type "struct ifreq" has member "ifr_hwaddr" : YES 
Checking for size of "long" : 8 
Program perl found: YES (/usr/bin/perl)
Program python3 found: YES (/usr/bin/python3)
Program xmllint found: YES (/usr/bin/xmllint)
Program xsltproc found: YES (/usr/bin/xsltproc)
Program rpcgen found: YES (/usr/bin/rpcgen)
Program augparse found: NO
Program dmidecode found: YES (/usr/bin/dmidecode)
Program ebtables found: YES (/usr/bin/ebtables)
Program flake8 found: NO
Program ip found: YES (/usr/bin/ip)
Program ip6tables found: YES (/usr/bin/ip6tables)
Program iptables found: YES (/usr/bin/iptables)
Program iscsiadm found: YES (/usr/bin/iscsiadm)
Program mdevctl found: NO
Program mm-ctl found: NO
Program modprobe found: YES (/usr/bin/modprobe)
Program ovs-vsctl found: NO
Program passt found: NO
Program pdwtags found: NO
Program rmmod found: YES (/usr/bin/rmmod)
Program scrub found: NO
Program tc found: YES (/usr/bin/tc)
Program udevadm found: YES (/usr/bin/udevadm)
Found pkg-config: /usr/bin/pkg-config (1.8.1)
Run-time dependency libtirpc found: YES 1.3.4
Run-time dependency libacl found: YES 2.3.1
Dependency libapparmor skipped: feature apparmor disabled
Library attr found: YES
Run-time dependency audit found: YES 3.1.2
Run-time dependency bash-completion found: YES 2.11
Run-time dependency blkid found: YES 2.39.2
Run-time dependency libcap-ng found: YES 0.8.3
Run-time dependency libcurl found: YES 8.4.0
Run-time dependency devmapper found: YES 1.02.196
Library dl found: YES
Check usable header "dlfcn.h" : YES 
Run-time dependency fuse3 found: YES 3.16.2
Run-time dependency glib-2.0 found: YES 2.78.1
Run-time dependency gobject-2.0 found: YES 2.78.1
Run-time dependency gio-unix-2.0 found: YES 2.78.1
Run-time dependency glusterfs-api found: YES 7.11.0
Run-time dependency gnutls found: YES 3.8.1
Run-time dependency libiscsi found: YES 1.19.0
Found CMake: /usr/bin/cmake (3.27.7)
Run-time dependency libnbd found: NO (tried pkgconfig and cmake)

libvirt-9.9.0/meson.build:1009:4: ERROR: Problem encountered: nbdkit support requires libnbd

A full log can be found at /var/tmp/pamac-build-pkehl/libvirt-xen/src/build/meson-logs/meson-log.txt
==> ERROR: A failure occurred in build().
    Aborting...
```

However, I get same error (from `pamac` even after I installed `nbdkit` explicitly.

![image](https://github.com/ahmubashshir/pkgbuilds/assets/4270240/1418dab5-55c4-4a11-8abd-8bec441ef91e)

![image](https://github.com/ahmubashshir/pkgbuilds/assets/4270240/ab6ab9d3-4120-4559-bd44-46acdeaf7b96)

Potentially a side note:
```
sudo pamac install libvirt-xen
...
==> Validating source files with sha256sums...
    libvirt-9.9.0.tar.xz ... Passed
    libvirt-9.9.0.tar.xz.asc ... Skipped
==> Verifying source file signatures with gpg...
    libvirt-9.9.0.tar.xz ... cat: write error: Broken pipe
FAILED
==> ERROR: One or more PGP signatures could not be verified!
Finished with result: exit-code
Main processes terminated with: code=exited/status=1
Service runtime: 16.973s
CPU time consumed: 1.364s
Error: Failed to build libvirt-xen
```

Only after I've manually installed `libnbd`, "Add/Remove" software succeeded for `libvirt-xen`.
![image](https://github.com/ahmubashshir/pkgbuilds/assets/4270240/67b65396-cba2-4255-8889-1569d2571fb4)

![image](https://github.com/ahmubashshir/pkgbuilds/assets/4270240/d7905538-b1b4-40b0-957d-98df2340ce67)

**Comment by ahmubashshir** [comment]
First of all, sorry for the awfully late response. I was busy with my semester finals.

I'm building the package right now (the CI needs a good rewrite). and the problem with nbd should be solved in v9.10.0

**Comment by peter-kehl** [comment]
Thank you.
Confirming that the newer (current) version builds fine.
